authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/DataShareManager.java

```diff
@@ -54,10 +54,13 @@ public class DataShareManager {
 	public <R> R downloadObject(String dataShareUrl, Class<R> clazz, boolean decryptionRequired) throws  RestServiceException, IdAuthenticationBusinessException {
 		RestRequestDTO request = restRequestFactory.buildRequest(RestServicesConstants.DATA_SHARE_GET, null, String.class);
 		request.setUri(dataShareUrl);
-		String responseStr = restTemplate.getForObject(dataShareUrl, String.class);
+		String responseStr = null;
+		if(restTemplate!=null){
+			responseStr = restTemplate.getForObject(dataShareUrl, String.class);
+		}
 		Optional<Entry<String, Object>> errorOpt = RestUtil.getError(responseStr, mapper);

```

Comment:
there is possibility of responseStr string object will be null in case the if statement is not executed.

Reply:
Corrected